src/hotspot/cpu/s390/assembler_s390.inline.hpp

```diff
@@ -1364,6 +1405,27 @@ inline unsigned int Assembler::instr_len(unsigned char *instr) {
   }
 }
 
+// Instruction must be stored right-justified in argument.
+inline unsigned int Assembler::instr_len(unsigned long instr) {
+  int len_bits = instr >> (48-2);
+  if (len_bits == 0) len_bits = instr >> (32-2);
+  if (len_bits == 0) len_bits = instr >> (16-2);
+  assert(len_bits < 4, "bad instruction %ld", instr);
+  switch (len_bits) {
+    case 0: return 2;
+    case 1: // fallthru
+    case 2: return 4;
+    case 3: return 6;
+    default:
+      // Control can't reach here.
+      // The switch expression examines just the leftmost two bytes
+      // of the main opcode. So the range of values is just [0..3].
+      // Having a default clause makes the compiler happy.
```

Comment:
A bit lengthy for a trivial explanation.

Reply:
You mean I should not explain what's going on?

Reply:
The explanation in the middle is fine. What's the value of the first and last sentence? The first one is already expressed by ShouldNotReachHere, the last one is a general statement which I don't like to be at every switch.

Reply:
Will rework with next commit.